rasa_core/tracker_store.py

```diff
@@ -250,3 +256,137 @@ def retrieve(self, sender_id):
 
     def keys(self):
         return [c["sender_id"] for c in self.conversations.find()]
+
+
+class SQLTrackerStore(TrackerStore):
+    """Store which can save and retrieve trackers from an SQL database"""
+
+    from sqlalchemy.ext.declarative import declarative_base
+    Base = declarative_base()
+
+    class SQLEvent(Base):
+        __tablename__ = 'conversations'
+
+        id = Column(Integer, primary_key=True)
+        sender_id = Column(String, nullable=False)
+        type_name = Column(String, nullable=False)
+        timestamp = Column(Float)
+        intent_name = Column(String)
+        action_name = Column(String)
+        data = Column(String)
+
+    def __init__(self,
+                 domain: Optional[Domain] = None,
+                 dialect: Text = 'sqlite',
+                 host: Text = None,
+                 event_broker: Optional[EventChannel] = None,
+                 db: Text = 'rasa.db',
+                 username: Text = None,
+                 password: Text = None) -> None:
+        from sqlalchemy.orm import sessionmaker
+        from sqlalchemy.engine.url import URL
+        from sqlalchemy import create_engine
+
+        engine_url = URL(dialect, username, password, host, database=db)
+
+        logger.debug('Attempting to connect to database '
+                     'via "{}"'.format(engine_url.__to_string__()))
+
+        self.engine = create_engine(engine_url)
+        self.session = sessionmaker(bind=self.engine)()
+
+        logger.debug('Connection successful, ensuring conversations table...')
+
+        self._ensure_table()
+        super(SQLTrackerStore, self).__init__(domain, event_broker)
+
+        logger.debug('SQL tracker store successfully initialised')
+
+    def _ensure_table(self) -> None:
+        """Creates the conversations table in the database if not present"""
+
+        metadata = MetaData()
+
+        Table("conversations", metadata,
```

Comment:
do we need these field definitions twice? I mean they are already part of the `SQLEvent ` class

Reply:
Managed to remove the `_ensure_table` method